File: python/exercises/inheritance/GeometricProgression.py
from Progression import Progression

class GeometricProgression(Progression):
    """Iterator producing a geometric progression."""

    def __init__(self, base=2, start=1):
        """Create a new GeometricProgression.
        base    the fixed constant multiplied to each term (default 2)
        start   the first term of the progression(default 1)
        """
        super().__init__(start)
        self._base = base

    def _advance(self):
        """ Overide inherited version.
        Update current value by multiplying by base value."""
        self._current *= self._base

File: java/exercises/inheritance/GeometricProgression.java
/** Generates geometric progression */
public class GeometricProgression extends Progression {

  // instance variable
  protected long base;

  /** Construct progression 1, 2, 4, .. */
  // start at 1 with base 2
  public GeometricProgression() { this(2, 1); }

  /** Construct progression 1, b, b^2, b^3, .... */
  public GeometricProgression(long b) { this(b, 1); }

  /** Construct progression with arbitary base and start */
  public GeometricProgression(long b, long start) {
    super(start);
    base = b;
  }

  /** Multiple the current value by the geometric base */
  protected void advance() {
    current *= base;
  }
}
